InformationScripting/src/queries/AstQuery.cpp

```diff
@@ -332,4 +405,87 @@ void AstQuery::adaptOutputForRelation(TupleSet& tupleSet, const QString& relatio
 	}
 }
 
+bool AstQuery::matchesExpectedType(Model::Node* node, Model::Node::SymbolType symbolType,
+											  const QString& expectedType, const QStringList& args)
+{
+	using SymbolType = Model::Node::SymbolType;
+	if (SymbolType::VARIABLE == symbolType)
+	{
+		if (auto varDecl = DCast<OOModel::VariableDeclaration>(node))
+			return stringifyType(varDecl->typeExpression()) == expectedType;
+	}
+	else
+	{
+		if (auto methodDecl = DCast<OOModel::Method>(node))
+		{
+			if (args.size() != methodDecl->arguments()->size()) return false;
+			// TODO: here we support only one return value:
+			if ((methodDecl->results()->size() == 0 && (expectedType.isEmpty() || expectedType == "void"))
+				 || (methodDecl->results()->size() > 0 &&
+					  stringifyType(methodDecl->results()->at(0)->typeExpression()) == expectedType)
+				 || expectedType == "_")
+			{
+				// return type matches check arguments:
+				bool matches = true;
+				for (int i = 0; i < args.size() && matches; ++i)
+				{
+					if (args[i] == "_") continue;
+					matches = stringifyType(methodDecl->arguments()->at(i)->typeExpression()) == args[i];
+				}
+				return matches;
+			}
+		}
+	}
+	return false;
+}
+
+QString AstQuery::stringifyType(OOModel::Expression* expression)
```

Comment:
Hmm, now that I look at this, I am thinking: can't you use the `StringComponents::stringForNode` to get this stringification?